app/providers/atmosphere/appliance_params.rb

```diff
@@ -0,0 +1,86 @@
+module Atmosphere
+  class ApplianceParams
+    def initialize(params, allowed_params_ext, delegated_auth)
+      @params = params
+      @allowed_params_ext = allowed_params_ext
+      @delegated_auth = delegated_auth
+    end
+
+    def create_params
+      (development? ? dev_params : prod_params).tap do |p|
+        at = config_template.appliance_type
+
+        p[:appliance_type_id] = at.id
+        p[:name] ||= at.name
+        p[:description] ||= at.description
+        p[:compute_sites] = allowed_compute_sites
+      end
+    end
+
+    def dev_mode_params
+      if development?
+        prefs = params.permit(dev_mode_property_set: [
+                                :preference_memory,
+                                :preference_cpu,
+                                :preference_disk
+                              ])
```

Comment:
Indent the right bracket the same as the start of the line where the left bracket is.